stm32-metapac/gen/src/lib.rs

```diff
@@ -267,12 +267,13 @@ pub fn gen(options: Options) {
                     _ => {}
                 }
 
-                if let Some(clock) = &p.clock {
-                    if let Some(rcc) = &rcc {
+                if let Some(rcc) = &rcc {
+                    let mut generate_rcc_peripheral = |clock_prefix| {
```

Comment:
Writing this with a closure is super weird :) You could do it with a var, like this:

```rust
let clock_prefix: &str = p.clock.as_ref().unwrap_or("");
// code from the closure here
```

Reply:
You're right, it is super weird. I can't remember why I did that, but it must have been because I thought of calling it directly from each peripheral type "spi" etc. and then I decided to do the "TIM hack" instead. I'll change it.